### PR TITLE
netbeans: 12.1 -> 12.2

### PR DESCRIPTION
**Pull request by asbachb**
###### Motivation for this change
Update to next minor release.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by bennyandresen — APPROVED** [review]
LGTM.
Builds under Sandbox and I was able to launch it.

/status needs_merger

**Review by timokau — APPROVED** [review]
Thanks!

File: pkgs/applications/editors/netbeans/default.nix [hunk 1]
```diff
@@ -19,14 +19,16 @@ stdenv.mkDerivation {
...
-    sha512 = "ad4bb5b191c784ed144b0b4831a8b95e0707c362917833c279d3f6fad11d7b3fb1f004f30121a941b694fc2ce323974b15072aa31cb5449111bc5d33d0d77103";
+    sha512 = "b25cda9830e8fe1d05687b08cc5fa9bcac7e8e6d12776998a4da7e483b3be0d04493345e56be7e6198fa8f86428d57d4459bfa7372c2e3f918f4a1101d0a31a7";
   };
 
   buildCommand = ''
     # Unpack and perform some path patching.
     unzip $src
     patchShebangs .
 
+    rm netbeans/bin/*.exe
```
Comment (timokau) [inline pkgs/applications/editors/netbeans/default.nix:30]: Its generally helpful to add a short comment that explains the "why" for something like this. In this case it's not important enough for another review iteration though.

**Pull request merged**